builder/dockerfile/internals.go

```diff
@@ -498,18 +498,22 @@ func (b *Builder) create() (*daemon.Container, error) {
 	}
 	b.runConfig.Image = b.image
 
+	resources := runconfig.Resources{
+		CPUShares:  b.CPUShares,
+		CPUPeriod:  b.CPUPeriod,
+		CPUQuota:   b.CPUQuota,
+		CpusetCpus: b.CPUSetCpus,
+		CpusetMems: b.CPUSetMems,
+		Memory:     b.Memory,
+		MemorySwap: b.MemorySwap,
+		Ulimits:    b.Ulimits,
+	}
+
 	// TODO: why not embed a hostconfig in builder?
 	hostConfig := &runconfig.HostConfig{
-		CPUShares:    b.CPUShares,
-		CPUPeriod:    b.CPUPeriod,
-		CPUQuota:     b.CPUQuota,
-		CpusetCpus:   b.CPUSetCpus,
-		CpusetMems:   b.CPUSetMems,
 		CgroupParent: b.CgroupParent,
```

Comment:
moving the container to a different cgroup which has some other constraints maybe, I'm moving it